thriftbp/errors.go

```diff
@@ -59,3 +59,13 @@ func BaseplateErrorFilter(codes ...int32) retrybp.Filter {
 		return next(err)
 	}
 }
+
+// NewBaseplateError is a helper function for creating baseplate.Error thrift objects.
+func NewBaseplateError(code baseplatethrift.ErrorCode, message string) *baseplatethrift.Error {
+	asInt32 := int32(code)
+	return &baseplatethrift.Error{
+		Code:    &asInt32,
```

Comment:
Why is the type defined as `optional i32` instead of `optional ErrorCode` in thrift :shakes-fist:

This actually can be inlined as `thrift.Int32Ptr(int32(code))`.

Reply:
So devs can add their own error codes.